File: version/api/slemp_api.py
# coding: utf-8

#------------------------------
# API-Demo of Python
#------------------------------
import time
import hashlib
import sys
import os
import json


class slempApi:
    __SLEMP_KEY = 'j7GQhzNcBV4KU9QKYPXvtjSzCcmfkc0e'
    __SLEMP_PANEL = 'http://127.0.0.1:7200'

    def __init__(self, slemp_panel=None, slemp_key=None):
        if slemp_panel:
            self.__SLEMP_PANEL = slemp_panel
            self.__SLEMP_KEY = slemp_key

    def __get_md5(self, s):
        m = hashlib.md5()
        m.update(s.encode('utf-8'))
        return m.hexdigest()

    def __get_key_data(self):
        now_time = int(time.time())
        ready_data = {
            'request_token': self.__get_md5(str(now_time) + '' + self.__get_md5(self.__SLEMP_KEY)),
            'request_time': now_time
        }
        return ready_data

    def __http_post_cookie(self, url, p_data, timeout=1800):
        cookie_file = './' + self.__get_md5(self.__SLEMP_PANEL) + '.cookie'
        if sys.version_info[0] == 2:
            # Python2
            import urllib
            import urllib2
            import ssl
            import cookielib

            cookie_obj = cookielib.MozillaCookieJar(cookie_file)

            if os.path.exists(cookie_file):
                cookie_obj.load(cookie_file, ignore_discard=True,
                                ignore_expires=True)

            ssl._create_default_https_context = ssl._create_unverified_context

            data = urllib.urlencode(p_data)
            req = urllib2.Request(url, data)
            opener = urllib2.build_opener(
                urllib2.HTTPCookieProcessor(cookie_obj))
            response = opener.open(req, timeout=timeout)

            cookie_obj.save(ignore_discard=True, ignore_expires=True)
            return response.read()
        else:
            # Python3
            import urllib.request
            import ssl
            import http.cookiejar
            cookie_obj = http.cookiejar.MozillaCookieJar(cookie_file)
            cookie_obj.load(cookie_file, ignore_discard=True,
                            ignore_expires=True)
            handler = urllib.request.HTTPCookieProcessor(cookie_obj)
            data = urllib.parse.urlencode(p_data).encode('utf-8')
            req = urllib.request.Request(url, data)
            opener = urllib.request.build_opener(handler)
            response = opener.open(req, timeout=timeout)
            cookie_obj.save(ignore_discard=True, ignore_expires=True)
            result = response.read()
            if type(result) == bytes:
                result = result.decode('utf-8')
            return result

    def getLogs(self):
        url = self.__SLEMP_PANEL + '/api/firewall/get_log_list'

        post_data = self.__get_key_data()
        post_data['limit'] = 10
        post_data['p'] = '1'

        result = self.__http_post_cookie(url, post_data)

        return json.loads(result)


if __name__ == '__main__':
    api = slempApi()

    rdata = api.getLogs()

    print(rdata)

File: version/api/slemp_api.php
<?php

class slempApi {
    private $SLEMP_KEY   = "j7GQhzNcBV4KU9QKYPXvtjSzCcmfkc0e";
    private $SLEMP_PANEL = "http://127.0.0.1:7200";

    public function __construct($slemp_panel = null, $slemp_key = null) {
        if ($slemp_panel) {
            $this->SLEMP_PANEL = $slemp_panel;
        }

        if ($slemp_key) {
            $this->SLEMP_KEY = $slemp_key;
        }
    }

    private function httpPostCookie($url, $data, $timeout = 60) {
        $cookie_file = './' . md5($this->SLEMP_PANEL) . '.cookie';
        if (!file_exists($cookie_file)) {
            $fp = fopen($cookie_file, 'w+');
            fclose($fp);
        }

        $ch = curl_init();
        curl_setopt($ch, CURLOPT_URL, $url);
        curl_setopt($ch, CURLOPT_TIMEOUT, $timeout);
        curl_setopt($ch, CURLOPT_POST, 1);
        curl_setopt($ch, CURLOPT_POSTFIELDS, $data);
        curl_setopt($ch, CURLOPT_COOKIEJAR, $cookie_file);
        curl_setopt($ch, CURLOPT_COOKIEFILE, $cookie_file);
        curl_setopt($ch, CURLOPT_RETURNTRANSFER, 1);
        curl_setopt($ch, CURLOPT_HEADER, 0);
        curl_setopt($ch, CURLOPT_SSL_VERIFYHOST, false);
        curl_setopt($ch, CURLOPT_SSL_VERIFYPEER, false);
        $output = curl_exec($ch);
        curl_close($ch);
        return $output;
    }

    private function getKeyData() {
        $now_time   = time();
        $ready_data = array(
            'request_token' => md5($now_time . '' . md5($this->SLEMP_KEY)),
            'request_time'  => $now_time,
        );
        return $ready_data;
    }

    public function getLogsList() {
        $url = $this->SLEMP_PANEL . '/api/firewall/get_log_list';

        $post_data          = $this->getKeyData();
        $post_data['p']     = '1';
        $post_data['limit'] = 10;

        $result = $this->httpPostCookie($url, $post_data);

        $data = json_decode($result, true);
        return $data;
    }

}

$api = new slempApi();
$rdata = $api->getLogsList();

// var_dump($rdata);
echo json_encode($rdata);

?>
